pkg/proc/native/proc.go

```diff
@@ -266,11 +266,23 @@ func (dbp *nativeProcess) initialize(path string, debugInfoDirs []string) (*proc
 		stopReason = proc.StopAttached
 	}
 	tgt, err := proc.NewTarget(dbp, dbp.pid, dbp.memthread, proc.NewTargetConfig{
-		Path:                path,
-		DebugInfoDirs:       debugInfoDirs,
-		DisableAsyncPreempt: runtime.GOOS == "windows" || runtime.GOOS == "freebsd",
-		StopReason:          stopReason,
-		CanDump:             runtime.GOOS == "linux"})
+		Path:          path,
+		DebugInfoDirs: debugInfoDirs,
+
+		// We disable asyncpreempt for the following reasons:
+		//  - on Windows asyncpreempt is incompatible with debuggers, see:
+		//    https://github.com/golang/go/issues/36494
+		//  - freebsd's backend is generally broken and asyncpreempt makes it even more so, see:
```

Comment:
Generally broken? I haven't seen an influx of failures on CI or anything.

Reply:
We have had a lot of important tests disabled on freebsd ever since it was merged, that's why.

Reply:
Ack, that's right.